### ISSUE TITLE
Misleading Log Output for `max_steps` in TFT and NHiTS

### ISSUE THREAD

**Issue by burton030berlin** [Nixtla/neuralforecast#1235]
### What happened + What you expected to happen

Hi everyone,

I've been using the implemented Temporal Fusion Transformer (TFT) and NHiTS models. I observed that when setting `max_steps = n`, the output log prints `Epoch n`.

From the documentation, my understanding is that `max_steps` refers to the number of gradient steps, not epochs. This behavior is confusing because it suggests that the training has progressed through `n` epochs, which doesn't align with my understanding.

Could you please confirm:

Is my understanding of `max_steps` (as gradient steps, not epochs) correct?
If so, is the log output incorrect and should it display "Step n" instead of "Epoch n"?
Let me know if additional information is needed to clarify or reproduce this issue.

Thanks for your help!

### Versions / Dependencies

neuralforecast 1.7.5, Python 3.9

Distributor ID:	Ubuntu
Description:	Ubuntu 22.04.5 LTS
Release:	22.04
Codename:	jammy

### Reproduction script

```
import pandas as pd
import matplotlib.pyplot as plt
import numpy as np
from neuralforecast import NeuralForecast
from neuralforecast.models import TFT
from neuralforecast.losses.pytorch import DistributionLoss
from neuralforecast.utils import AirPassengersPanel, AirPassengersStatic

AirPassengersPanel['month']=AirPassengersPanel.ds.dt.month
Y_train_df = AirPassengersPanel[AirPassengersPanel.ds<AirPassengersPanel['ds'].values[-12]] # 132 train
Y_test_df = AirPassengersPanel[AirPassengersPanel.ds>=AirPassengersPanel['ds'].values[-12]].reset_index(drop=True) # 12 test

nf = NeuralForecast(
    models=[TFT(h=12, input_size=48,
                hidden_size=20,
                loss=DistributionLoss(distribution='StudentT', level=[80, 90]),
                learning_rate=0.005,
                stat_exog_list=['airline1'],
                futr_exog_list=['y_[lag12]','month'],
                hist_exog_list=['trend'],
                max_steps=10,
                scaler_type='robust',
                windows_batch_size=None,
                enable_progress_bar=True),
    ],
    freq='M'
)

nf.fit(df=Y_train_df, static_df=AirPassengersStatic, val_size=12)
```

### Issue Severity

Medium: It is a significant difficulty but I can work around it.

**Comment by marcopeix** [comment]
Hello! Your understanding is correct and you can see the relationship between steps and epochs here: #600

**Comment by jmoralez** [comment]
Both the epoch and the steps are shown in the progress bar. Here's an example where I have 2k unique_ids and set batch_size=32 and max_steps=100. You can see that each epoch has 63 steps (2,000 / 32) and it trained for 1 epoch + 37 steps = 63 + 37 = 100 steps.

![image](https://github.com/user-attachments/assets/f9e879a7-fc1d-4956-90fe-b889d33de7d9)

**Comment by burton030berlin** [comment]
@marcopeix Thanks for the link. 
But why in the screenshot below it shows 100 epochs when setting `max_step=100` and `batch_size=32` while training examples are 264?

![image](https://github.com/user-attachments/assets/19c7d1cb-f062-47d0-b9dc-568cb2f8a599)
